src/v2/guide/typescript.md

```diff
@@ -0,0 +1,104 @@
+---
+title: TypeScript
+type: guide
+order: 25
+---
+
+If you develop a large application, using a language having static type checking would be helpful. If your code seems to cause unexpected behavior, it provides useful errors before you run it on browsers. You can easily use [TypeScript](https://www.typescriptlang.org/) (one of popular AltJS having static type checking) with Vue since Vue officially provides its declaration files.
+
+## Official Declaration Files
+
+There are official declaration files for Vue, vue-router and vuex. You do not have to use external tools like `typings` to use them since they are published via npm. You can install Vue's declarations with `npm install vue`. Then, all declarations can be imported with `import` statement.
+
+Example of component definition with TypeScript:
+
+``` ts
+// we cannot write `import Vue from 'vue'` in TypeScript
+// because of the difference of module resolution strategy
+// against the other AltJS and module bundlers
+import Vue = require('vue')
+
+Vue.component('my-component', {
+  template: '<div>A custom component!</div>'
+})
+```
+
+Note that `Vue` is already typed, so all methods, properties and parameters will be type checked. For example, if you typo `template` option, TypeScript compiler will print an error message on a compile time.
+
+If you want to use Vue's types explicitly, they belong to the `Vue` object. In case you create a component on a separated file (e.g. [single file component](single-file-components.html)), you can annotate a type with the component option object:
+
+``` ts
+import Vue = require('vue')
+
+export default {
+  props: ['message'],
+  template: '<span>{{ message }}</span>'
+
+// annotate ComponentOptions with `as` operator
+} as Vue.ComponentOptions<Vue>
+```
+
+You can see all available types on the corresponding repository ([here are Vue's types](https://github.com/vuejs/vue/blob/dev/types/index.d.ts)).
+
+<p class="tip">The declaration files of Vue requires `--lib DOM,ES2015` as compiler option. So you need to pass the option via `tsc` command or `tsconfig.json`. For more details of compiler options, please see [Compiler Options · TypeScript](https://www.typescriptlang.org/docs/handbook/compiler-options.html)</p>
```

Comment:
I would suggest to require only `--lib DOM,ES2015.promise`. Because `Promise` is the only requirements in Vue(and it is optional, if users do not use async component.). IMHO, listing only API we used is in accordance with the rule of least power.